src/Microsoft.AspNetCore.Sockets.Client.Http/Constants.cs

```diff
@@ -1,13 +1,25 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System.Linq;
 using System.Net.Http.Headers;
+using System.Reflection;
 
 namespace Microsoft.AspNetCore.Sockets.Client.Http
 {
-    public class Constants
+    public static class Constants
     {
-        private static readonly string UserAgent = "Microsoft.AspNetCore.Sockets.Client.Http/1.0.0-alpha";
-        public static readonly ProductInfoHeaderValue UserAgentHeader = ProductInfoHeaderValue.Parse(UserAgent);
+        public static readonly ProductInfoHeaderValue UserAgentHeader;
+
+        static Constants()
+        {
+            var assemblyVersion = (AssemblyInformationalVersionAttribute)typeof(Constants)
+                .Assembly
+                .GetCustomAttributes(typeof(AssemblyInformationalVersionAttribute))
+                .Single();
```

Comment:
SingleOrDefault? I guess this should never fail but this sorta thing makes me paranoid

Reply:
I'm not sure of a world where that could happen. Plus the unit tests are doing the same check.

What should the version be if the attribute couldn't be found?

Reply:
`Microsoft.AspNetCore.Sockets.Client.Http/unknown`?

Reply:
https://tools.ietf.org/html/rfc7231#section-5.5.3

Spec says version in user-agent is optional. I'll not include it if for some crazy reason the attribute can't be found.

Reply:
Add a Debug.Assert that it's never null. That should be fine.